pkg/pillar/diskmetrics/diskmetrics.go

```diff
@@ -63,3 +64,20 @@ func ResizeImg(diskfile string, newsize uint64) error {
 	}
 	return nil
 }
+
+// CreateScratchDisk will create a disk of given format and size
+func CreateScratchDisk(diskfile, format string, size uint64) error {
```

Comment:
Why does this function belong to diskmetrics.go ? Doesn't seem like the right place for it.

Also, what is the proposed end-to-end use for it?

Reply:
It is a pre-work to add the functionality of creating scratch disks in EVE. The user will give the details of the scratch disk like format, size, etc. in manifest instead of uploading it and EVE will create a disk and mount it in the instance.
If HDD_EMPTY is set in the manifest, EVE will bypass the downloader and verifier.

Reply:
Is utils the right place for it?

Reply:
Honestly, the right place for it is the new service called volumemgr that we should start implementing ASAP. Most of the functionality for that service is going to come from domainmanager so I suggest you put this there for now.

As for e2e scratch disk -- I'll let @kalyan-nidumolu comment on what's the best way to sequence this with the other work on volumes that's in-flight so you guys don't have to do double work.